### ISSUE TITLE
pypy3 didn't support ctypes bit-fields

### ISSUE THREAD

**Issue by mlouielu** [iovisor/bcc#2416]
https://github.com/iovisor/bcc/blob/35cb9fd1cb0884e7a1fed1d3c01314655e40482c/src/python/bcc/disassembler.py#L26-L33

`BPFInstrFields` create fields with bit-fields, causing pypy binding failed. Not sure how to fixed it. I do a workaround with remove bit-fields and it works fine (in my case). Will it cause some issue?

Thanks.

**Comment by yonghong-song** [comment]
In this particular case, there is no issue. Originally, the bit fields are added mostly for __int128 type where there is no corresponding types in python ctype.

Could you help print out what kinds of bitfields are added by bcc? In this case, it looks like bitfield should not be added.

**Comment by mlouielu** [comment]
@yonghong-song How do I print it out? Thanks.

**Comment by yonghong-song** [comment]
@mlouielu I guess I may misunderstand your original questions as I am not really familiar with pypy. Now I know what it is (an alternative to python).
So you want to remove the two bitfields, dst and src, below
```
     _fields_ = [('opcode', ct.c_uint8), 
                 ('dst', ct.c_uint8, 4), 
                 ('src', ct.c_uint8, 4), 
                 ('o', OffsetUnion), 
                 ('i', ImmUnion)] 
```
for pypy.
Is this what you want?

These two fileds are used later in the same file (disassembler.py) and also this structure accurately represents the kernel data structure, I hesitate to change it.

**Comment by mlouielu** [comment]
Thanks for the elaboration, yes, this is what I ask for. I know it should be used in else where, are there any test case for this? So I can test it with pypy for it.

**Comment by yonghong-song** [comment]
Yes, python test test_disassembler.py is for this functionality.
